docs/proposals/koordlet/20230728-support-cold-memory-compute.md

```diff
@@ -0,0 +1,245 @@
+	Cfea                []uint64 `json:"cfea"`
+	Dfea                []uint64 `json:"dfea"`
+	Csua                []uint64 `json:"csua"`
+	Dsua                []uint64 `json:"dsua"`
+	Cfua                []uint64 `json:"cfua"`
+	Dfua                []uint64 `json:"dfua"`
+	Slab                []uint64 `json:"slab"`
+}
+```
+
+Define a function named readidleInfo() in koordlet util moudle. It can read idleinfo memory.idle_stat which exists  under every hierarchy of cgroup memory. (node e.g. /sys/fs/cgroup/memory/memory.idle_stat)
+
+```go
+func readIdleInfo(path string) (*IdleInfo, error) 
+```
+
+Define a function named GetIdleInfoFilePath(). It is uesd to return idleinfo file path.(node e.g. /sys/fs/cgroup/memory/memory.idle_stat)
+
+```go
+func GetIdleInfoFilePath(idleFileRelativePath string) string 
+```
+
+Define a function named GetIdleMemoryTotalKB(). It is uesd to return size of total idle page. The unit is byte.
+
+```go
+func GetIdleMemoryTotalKB(idleFileRelativePath string) (uint64, error) 
+```
+
+Define two functions  in pkg/koordlet/util/meminfo.go  to get node's MemTotal and MemFree. The values are used to compute Mem usage with hot page.
+
+```go
+// GetMemTotalKB returns the node's memory total quantity (kB)
+func GetMemTotalKB() (int64, error) 
+// GetMemFreeKB returns the node's memory free quantity (kB)
+func GetMemFreeKB() (int64, error)
+```
+
+Define a function named collectNodeMemoryWithHotPageUsage().  Compute node memory usage with hot page and generate metric. collectNodeResUsed() function will call that and insert memoryWithHotPage metric.
+
+```go
+func (n *nodeResourceCollector) collectNodeMemoryWithHotPageUsage(collectTime time.Time) (metriccache.MetricSample, error) 
+```
+
+Define a function named collectNodeMemoryColdPageSize().  Compute node memory cold page size and generate metric. collectNodeResUsed() function will call that and insert memoryColdPageSize metric.
+
+```go
+func (n *nodeResourceCollector) collectNodeMemoryColdPageSize(collectTime time.Time) (metriccache.MetricSample, error)
+```
+
+
+Add a new resource collector. Define a coldMemoryResouceCollector to manager all coldPageCollectors  and implement collector interface.
+```go
+type coldMemoryResouceCollector struct {
+	collectInterval time.Duration
+	started         *atomic.Bool
+	supported       *atomic.Bool
+	appendableDB    metriccache.Appendable
+	metricDB        metriccache.MetricCache
+	//concrete cold page collectors sush as kidled, kstaled and DAMON、
+	coldPageCollectors map[string]framework.Collector
+}
+func (c *coldMemoryResouceCollector) Enabled() bool 
+func (c *coldMemoryResouceCollector) Setup(c *framework.Context) 
+func (c *coldMemoryResouceCollector) Run(stopCh <-chan struct{}) 
+func (c *coldMemoryResouceCollector) Started() bool
+func (c *coldMemoryResouceCollector) IsSupported() bool 
+```
+
+Define a concrete a coldPageCollector called kidled to cold page info of node, pod and container.
```

Comment:
Perhaps we do not need to define two collectors for the cold page feature. How about defining one cold page collector and leaving the different kernel implementations inside this collector?